### PR TITLE
Refactor firestore modifiers and gtag events

### PR DESCRIPTION
**Pull request by benjamin-shen**
### Summary 

This pull request cleans up some of the data modifier code after the VueX migration.

- [x] refactor firestore modifiers to be directly called from `global-firestore-data.ts` (opposed to having convoluted and repeated logic in the components)
- [x] refactor firestore-related gtag events from the semester components to `global-firestore-data.ts`
- [x] refactor all gtag events; they are now handled in `gtag.ts`

Remaining TODOs:

- [ ] Test that gtag still works properly

### Test Plan 

Make sure add course, delete course, add semester, delete semester still work properly. Now that the requirements bar listens to firestore changes, the important thing to test is that the `user-semesters` data is being stored properly in firestore.

### Notes 

When using or working with firestore modifiers (or any global functions), we should try to move the function call closer to the source of the trigger. This is to try to clean up the long $emit chains when possible.

### CODE REVIEW

File: src/global-firestore-data.ts [hunk 1]
```diff
@@ -44,21 +37,62 @@ export const compareFirestoreSemesters = (a: FirestoreSemester, b: FirestoreSeme
...
+  courses: readonly FirestoreSemesterCourse[] = []
+): void => {
+  GTagEvent(gtag, 'add-semester');
+  editSemesters(oldSemesters =>
+    [...oldSemesters, createSemester(type, year, courses)].sort(compareFirestoreSemesters)
+  );
+};
+
+export const deleteSemester = (type: FirestoreSemesterType, year: number, gtag?: GTag): void => {
+  GTagEvent(gtag, 'add-semester');
```
Comment (SamChou19815) [inline src/global-firestore-data.ts:83]: typo?

**Review by SamChou19815 — APPROVED** [review]
Changes LGTM, with one potential issue

**Review by tcho6319 — CHANGES_REQUESTED** [review]
Tysm Ben! LGTM. I would fix the typo that Sam pointed out but other than that, this PR should be g2g

File: src/global-firestore-data.ts (hunk 1 above)
Comment (benjamin-shen) [reply src/global-firestore-data.ts]: Thanks lol good catch

**Review by SamChou19815 — APPROVED** [review]

**Review by tcho6319 — APPROVED** [review]

**Pull request merged**